static/app/components/modals/widgetViewerModal.tsx

```diff
@@ -173,14 +172,27 @@ function WidgetViewerModal(props: Props) {
 
   const [chartUnmodified, setChartUnmodified] = React.useState<boolean>(true);
 
-  const [modalSelection, setModalSelection] =
+  const [chartZoomOptions, setChartZoomOptions] = React.useState<DataZoomComponentOption>(
+    {start: 0, end: 100}
+  );
+
+  const [modalTableSelection, setModalTableSelection] =
     React.useState<PageFilters>(locationPageFilter);
+  const modalChartSelection = React.useMemo(() => modalTableSelection, [selection]);
```

Comment:
So `modalChartSelection` is only used when a user directly visits a widget viewer link. This is to set the initial `PageFilters` of the widget viewer (which determines the outer bounds of the minimap/slider). The outer bounds of the minimap shouldn't change after the initial render (which would cause a rerender, if it did change), which is why we're using `useMemo`. Although I've marked `selection` as a dependancy of this `useMemo`, it's not possible to change `selection` while the widget viewer is open, so the value of the `useMemo` is pretty much permanently fixed to the initial value (which is what we want).

I took a look at this again since you brought it up, and I've made a change to use `useRef` instead of `useMemo` since it seems more suitable for this use case. The value of `useRef` is never changed so it's functionally the same as before, but this time we don't need to specify any dependancies like we do with `useMemo`.